unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs

```diff
@@ -0,0 +1,573 @@
+#endif
+                    tempConfigLoaded = false;
+                    return tempConfigLoaded;
+                }
+            }
+
+            // Register to timelineEvents
+            if (configuration != null)
+            {
+                configuration.OnTimelineEvent -= Configuration_OnTimelineEvent;
+            }
+            newConfiguration.OnTimelineEvent += Configuration_OnTimelineEvent;
+            configuration = newConfiguration;
+
+            // Apply material as per number of Slots.
+            MaterialReferenceContainer.Mat_Layer matLayer = MaterialReferenceContainer.i.GetMat_LayerForLayers(configuration.slots.Count);
+            if (matLayer == null)
+            {
+                matLayer = MaterialReferenceContainer.i.materials[0];
+            }
+
+            configuration.ResetMaterial(matLayer.material, matLayer.numberOfSlots);
+            selectedMat = matLayer.material;
+            slotCount = matLayer.numberOfSlots;
+
+            if (DataStore.i.skyboxConfig.useProceduralSkybox.Get())
+            {
+                RenderSettings.skybox = selectedMat;
+            }
+
+            // Update loaded config
+            loadedConfig = configToLoad;
+
+            return tempConfigLoaded;
+        }
+
+        private void Configuration_OnTimelineEvent(string tag, bool enable, bool trigger) { OnTimelineEvent?.Invoke(tag, enable, trigger); }
+
+        // Update is called once per frame
+        public void Update()
+        {
+            if (!DataStore.i.skyboxConfig.disableReflection.Get() && skyboxProbe != null && !probeParented)
+            {
+                AssignCameraInstancetoProbe();
+            }
+
+            if (configuration == null || isPaused || !DataStore.i.skyboxConfig.useProceduralSkybox.Get())
+            {
+                return;
+            }
+
+            // Control is in editor tool
+            if (overrideByEditor)
+            {
+                return;
+            }
+
+            timeOfTheDay += Time.deltaTime / timeNormalizationFactor;
+            timeOfTheDay = Mathf.Clamp(timeOfTheDay, 0.01f, cycleTime);
+            DataStore.i.skyboxConfig.currentVirtualTime.Set(timeOfTheDay);
+
+            syncCounter++;
+
+            if (syncCounter >= syncAfterCount)
+            {
+                GetTimeFromTheServer(WorldTimer.i.GetCurrentTime());
+                syncCounter = 0;
+            }
+
+            configuration.ApplyOnMaterial(selectedMat, timeOfTheDay, GetNormalizedDayTime(), slotCount, directionalLight, cycleTime);
+
+            // Cycle resets
+            if (timeOfTheDay >= cycleTime)
+            {
+                timeOfTheDay = 0.01f;
+                configuration.CycleResets();
+            }
+
+        }
+
+        public void Dispose()
+        {
+
+            CommonScriptableObjects.proceduralSkyboxDisabled.Set(true);
+            CommonScriptableObjects.proceduralSkyboxEnabled.Set(false);
```

Comment:
Why do we have two bools for handling the disabled/enabled state? We should have only one.

Reply:
This is done as we have two types of controls in UI
1. Day/Night control which need proceduralSkyboxDisabled flag as true get deactivated
2. Procedural skybox controls: which need proceduralSkyboxDisabled flag as true to get deactivated

So, same boolean variable can't be used for both of these.